types/artifacts.d.ts

```diff
@@ -743,6 +741,56 @@ declare global {
         /** Column number in the script (0-based). */
         columnNumber: number;
       }
+
+      /** Describes a generic console message. */
+      interface BaseConsoleMessage {
+        /**
+         * What emitted the console message (console API, violation log, etc.).
+         * See subtypes for possible values.
+         */
+        source: string,
```

Comment:
I'm somewhat surprised to see these listed here given we want to narrow them later (TIL [it works just fine](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEJwM4oN4ChkHIAeAXMhmFKAOa4C+uuoksiKAYgPafIRGQgAJhjSYc+QqWQByGN2n1GzaPCSiovfhCEj0WZHkLEy0gEZwoChrjABPAA4du6DQF5kXHgB91jGAFcQBDBgThBkSAoACnMALzJPFwBKMgA3TmBBAwkCBDCMTgAbCAA6Qs5qGLhYkqIkxRsIaOwpaVtOaTokoA) 😮 ), but I do like the one place to get a central picture. 

Curious if any deeper thoughts here :)

Reply:
Yeah, it's totally not needed here but I thought it's clearer to put it out here. I'm happy to change if you have strong feelings.

TBH I wasn't sure if it'd work either but I tested it with [this](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN7JpgLmQGcwpQBzZAXwChbRJZEUBBZCAD0hABNi0mHHkLIA5HDHIAPuNZiA3MgBGcKEQmSZc+TQbho8JMgBCHbhD4CMWXPnQblU2WJOKVax0+2u3dMACeAA5sZgC8yOyyJgr0MACuIAhgwOggyEHk4AAUcMpErCYAlDi0yMgIacToADYQAHQ16BS5ykWxdLSZjNnY9o5iADQe6uIICGLURbRAA)